server/tools/build-keycloak.sh

```diff
@@ -50,17 +50,29 @@ fi
 
 mkdir -p /opt/jboss/keycloak/modules/system/layers/base/com/mysql/jdbc/main
 cd /opt/jboss/keycloak/modules/system/layers/base/com/mysql/jdbc/main
-curl -O https://repo1.maven.org/maven2/mysql/mysql-connector-java/$JDBC_MYSQL_VERSION/mysql-connector-java-$JDBC_MYSQL_VERSION.jar
+if [ -x "$(command -v mvn)" ]; then
+  mvn dependency:get -Dartifact=mysql:mysql-connector-java:$JDBC_MYSQL_VERSION:jar -Ddest=mysql-connector-java-$JDBC_MYSQL_VERSION.jar
+else
+  curl -O https://repo1.maven.org/maven2/mysql/mysql-connector-java/$JDBC_MYSQL_VERSION/mysql-connector-java-$JDBC_MYSQL_VERSION.jar
+fi
 cp /opt/jboss/tools/databases/mysql/module.xml .
 
 mkdir -p /opt/jboss/keycloak/modules/system/layers/base/org/postgresql/jdbc/main
 cd /opt/jboss/keycloak/modules/system/layers/base/org/postgresql/jdbc/main
-curl -L https://repo1.maven.org/maven2/org/postgresql/postgresql/$JDBC_POSTGRES_VERSION/postgresql-$JDBC_POSTGRES_VERSION.jar > postgres-jdbc.jar
+if [ -x "$(command -v mvn)" ]; then
+  mvn dependency:get -Dartifact=org/postgresql:postgresql:$JDBC_POSTGRES_VERSION:jar -Ddest=postgres-jdbc.jar
+else
+  curl -L https://repo1.maven.org/maven2/org/postgresql/postgresql/$JDBC_POSTGRES_VERSION/postgresql-$JDBC_POSTGRES_VERSION.jar > postgres-jdbc.jar
+fi
 cp /opt/jboss/tools/databases/postgres/module.xml .
 
 mkdir -p /opt/jboss/keycloak/modules/system/layers/base/org/mariadb/jdbc/main
 cd /opt/jboss/keycloak/modules/system/layers/base/org/mariadb/jdbc/main
-curl -L https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/$JDBC_MARIADB_VERSION/mariadb-java-client-$JDBC_MARIADB_VERSION.jar > mariadb-jdbc.jar
+if [ -x "$(command -v mvn)" ]; then
+  mvn dependency:get -Dartifact=org/mariadb/jdbc:mariadb-java-client:$JDBC_MARIADB_VERSION:jar -Ddest=mariadb-jdbc.jar
```

Comment:
Personally, I'm not a big fan of this approach. 

It adds additional complexity by introducing `if/else` blocks and it doesn't really buy us anything. It also increases overall build time as Maven checks also `maven-metadata.xml`.

For my personal taste, using `curl` and hardcoding Maven Central repo here is enough. But maybe others (@abstractj @stianst @iankko) like this approach.

Reply:
I'm against these changes as well. To solve the use-cases in this case I'd be happy to have the Maven repo set as a variable like we did for download repo.